anta/tests/connectivity.py

```diff
@@ -16,34 +27,64 @@ class VerifyReachability(AntaTest):
     Expected Results:
         * success: The test will pass if all destination IP(s) are reachable.
         * failure: The test will fail if one or many destination IP(s) are unreachable.
-        * error: The test will give an error if the destination IP(s) or the source interface/IP(s) are not provided as template_params.
     """
 
     name = "VerifyReachability"
     description = "Test the network reachability to one or many destination IP(s)."
     categories = ["connectivity"]
-    template = AntaTemplate(template="ping {dst} source {src} repeat 2")
+    commands = [
+        AntaTemplate(template="ping vrf {vrf} {destination} source {source} repeat 2"),
+        AntaTemplate(template="ping vrf {vrf} {destination} interface {interface} repeat 2"),
+    ]
 
-    @AntaTest.anta_test
-    def test(self) -> None:
-        """
-        Run VerifyReachability validation.
-        """
+    class Input(AntaTest.Input):  # pylint: disable=missing-class-docstring
+        hosts: List[Host]
+        """List of hosts to ping"""
 
-        failures = []
+        class Host(BaseModel):
+            """Remote host to ping"""
 
-        for command in self.instance_commands:
-            if command.params and ("src" and "dst") in command.params:
-                src, dst = command.params["src"], command.params["dst"]
-            else:
-                self.result.is_error("The destination IP(s) or the source interface/IP(s) are not provided as template_params")
-                return
+            destination: IPv4Address
+            """IPv4 address to ping"""
+            source: Optional[IPv4Address] = None
```

Comment:
Should support a source interface also.

Reply:
It does, see below.
However, I thought the previous test was wrong because of this:
```
dc1-leaf1a#ping vrf MGMT 1.1.1.1 source ?
  A.B.C.D          IP address
  A:B:C:D:E:F:G:H  IP address
```
And should instead use:
```
dc1-leaf1a#ping vrf MGMT 1.1.1.1 interface ?
  Ethernet      Ethernet Sub Interface
  Fabric        Fabric Interfaces
  Loopback      Hardware interface used for looping packets
  Management    Management interface
  Port-Channel  Link Aggregation Group (LAG)
  Tunnel        Tunnel interface
  Vlan          Logical interface into a VLAN
  Vxlan         VXLAN Tunnel Interface
 ```
 But this works:
 ```
 dc1-leaf1a#ping vrf MGMT 1.1.1.1 source Management0
PING 1.1.1.1 (1.1.1.1) from 192.168.100.129 : 72(100) bytes of data.
80 bytes from 1.1.1.1: icmp_seq=1 ttl=55 time=34.0 ms
80 bytes from 1.1.1.1: icmp_seq=2 ttl=55 time=29.1 ms
80 bytes from 1.1.1.1: icmp_seq=3 ttl=55 time=41.1 ms
80 bytes from 1.1.1.1: icmp_seq=4 ttl=55 time=37.9 ms
80 bytes from 1.1.1.1: icmp_seq=5 ttl=55 time=36.4 ms
```

So we can simplify the test and use Union and a single CLI command